src/browser_authentication_app_skeleton/src/actions/sign_ups/create.cr

```diff
@@ -1,9 +1,9 @@
 class SignUps::Create < BrowserAction
-  include Auth::RedirectIfSignedIn
+  include Auth::RedirectSignedInUsers
 
   route do
     SignUpForm.create(params) do |form, user|
-      if user
+      if form.valid? && user
```

Comment:
Here's another idea. Always check if the form is valid first. Then we can just do `yield self, user` without the conditional. I really haven't thought this through a ton. I'm not totally sure I like it, but thought I'd see what others think.

One downside if you now *always* have to check if the form is valid, and if you forgot and only check for user you might accidentally sign a user in! :S